tests/platform_helper/utils/test_validation.py

```diff
@@ -647,6 +648,47 @@ def test_load_and_validate_config_valid_file(yaml_file):
     assert validated == conf
 
 
+def test_lint_yaml_for_duplicate_keys_fails_when_duplicate_keys_provided(
+    valid_platform_config, fakefs
+):
+    fakefs.create_file(PLATFORM_CONFIG_FILE, contents=yaml.dump(valid_platform_config))
+    # Remove the extensions k:v pair from the platform config - re-added as plain text.
+    valid_platform_config.pop("extensions")
+
+    naughty_extensions_with_duplicate = """
+extensions:
+  connors-favourite-key-but-duplicated:
+    type: redis
+    environments:
+        "*":
+        engine: '7.1'
+        plan: tiny
+        apply_immediately: true
+  connors-favourite-key-but-duplicated:
+    type: redis
+    environments:
+      "*":
+          engine: '7.1'
+          plan: tiny
+          apply_immediately: true
+"""
+
+    # Kinda hacky - need to add the duplicate keys back into the platform-config as a string.
+    # As adding keys as a dictionary will overwrite the duplicate keys, hence no duplicate and the test doesn't work.
+
+    invalid_platform_config = f"""
+{yaml.dump(valid_platform_config)}
+{naughty_extensions_with_duplicate}
+"""
```

Comment:
Nice

Reply:
gave a bit more context in the comments, but do tell if something does not make sense. The formatting of the string is annoying, we can only keep it on the side, "squished" on the left. It doesn't look good but there's no alternative that I could find.